File: code/2390.removing-stars-from-a-string.php
<?php
/*
 * @lc app=leetcode id=2390 lang=php
 *
 * [2390] Removing Stars From a String
 */

// @lc code=start
class Solution {

    /**
     * @param String $s
     * @return String
     */
    function removeStars($s) {
        $result = [];
        for ($i = 0; $i < strlen($s); $i++) {
            if ($s[$i] === '*') {
                array_pop($result);
                continue;
            }
            $result[] = $s[$i];
        }

        return implode($result);
    }
}
// @lc code=end

File: code/2390.removing-stars-from-a-string.go
package code

/*
 * @lc app=leetcode id=2390 lang=golang
 *
 * [2390] Removing Stars From a String
 */

// @lc code=start
func removeStars(s string) string {
	stack := make([]byte, 0)

	for i := 0; i < len(s); i++ {
		if s[i] == '*' {
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, s[i])
		}
	}

	return string(stack)
}

// @lc code=end
